distribution.md

```diff
@@ -43,29 +44,219 @@ Each buildpack layer blob MUST contain a single buildpack at the following file
+
+#### Filesystem
+
+A builder MUST have the following directories/files:
+
+* `/cnb/order.toml` &rarr; As defined in the [platform specification](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml).
+
+#### Environment Variables
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+#### Labels
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
+
+A builder image MAY contain buildpacks, and MAY retain all the specified Labels set on the original buildpackage, as specified in the [buildpackage](#buildpackage) specification with the following exceptions:
 
-For each listed stack, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
+- `io.buildpacks.buildpack.metadata` MUST not be set.
+- `io.buildpacks.buildpack.layers`  on the builder SHOULD be a merged version based on all buildpackages combined and thereby have of all packaged buildpacks represented.
+
+In addition to all inherited labels above, the following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Label                            | Description                                                   |
+| -------------------------------- | ------------------------------------------------------------- |
+| `io.buildpacks.builder.metadata` | A JSON object representing builder metadata.                  |
+| `io.buildpacks.buildpack.order`  | A JSON object representation of the `/cnb/order.toml` file.   |
+
+##### `io.buildpacks.builder.metadata` (JSON)
+
+```json
+{
+  "description": "<string>",
+  "createdBy": {
+    "name": "<string>",
+    "version": "<string>",
+  }
+}
+```
+
+Where:
+
+- `description` (optional) is a human readable description of the builder.
+- `createdBy` (optional) is information about the tool that created the builder.
+  - `name` (optional) is the name of the tool that created the builder.
+  - `version` (optional) is the version of the tool that created the builder.
+
+## Data Format
+
+### Files
+
+#### `builder.toml` (TOML)
+```toml
+description = "a human readable description output when you run `pack inspect-builder`"
+
+[[buildpacks]]
+id = "<buildpack ID>"
+version = "<buildpack version>"
+uri = "<uri>"
+
+[[order]]
+[[order.group]]
+id = "<buildpack ID>"
+version = "<buildpack version>"
+optional = false
+
+[stack]
+id = "<stack id>"
+build-image = "build image"
+run-image = "run image"
+run-image-mirrors = "mirrors"
```

Comment:
```suggestion
```

I think we should omit this since we just adding the spec now

Reply:
I see your point,  but I think we're going to be in a fuzzy middle ground of "deprecated but still supported" for a while.   I've moved it out of the main section and under the specific bullet point that mentions that stack is deprecated,  so that it's co-located with its deprecation notice and a casual skimming of the docs wouldn't suggest it's part of the main spec.